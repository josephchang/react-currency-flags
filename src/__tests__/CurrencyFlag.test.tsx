import React from 'react';
import renderer, { ReactTestRenderer } from 'react-test-renderer';
import 'jest-styled-components';
import flags from '../flags';
import CurrencyFlag, { CurrencyFlagSizes, ICurrencyFlagProps } from '../';

describe('<CurrencyFlag />', () => {
  let props: ICurrencyFlagProps;
  let tree: ReactTestRenderer;

  describe('GIVEN a valid/supported currency', () => {
    const currency = 'cad';

    beforeEach(() => {
      props = { currency };
    });

    describe('AND a predefined size', () => {
      describe('AND the size `sm`', () => {
        const size = CurrencyFlagSizes.SMALL;

        beforeEach(() => {
          tree = renderer.create(<CurrencyFlag {...props} size={size} />);
        });

        it('should display the flag in `sm`', () => {
          expect(tree.toJSON()).toMatchSnapshot();
          expect(tree.toJSON()).toHaveStyleRule(
            'background-image',
            `url("${flags[currency]}")`
          );
        });
      });

      describe('AND the size `md`', () => {
        const size = CurrencyFlagSizes.MEDIUM;

        beforeEach(() => {
          tree = renderer.create(<CurrencyFlag {...props} size={size} />);
        });

        it('should display the flag in `md`', () => {
          expect(tree.toJSON()).toMatchSnapshot();
        });
      });

      describe('AND the size `lg`', () => {
        const size = CurrencyFlagSizes.LARGE;

        beforeEach(() => {
          tree = renderer.create(<CurrencyFlag {...props} size={size} />);
        });

        it('should display the flag in `lg`', () => {
          expect(tree.toJSON()).toMatchSnapshot();
        });
      });

      describe('AND the size `xl`', () => {
        const size = CurrencyFlagSizes.XLARGE;

        beforeEach(() => {
          tree = renderer.create(<CurrencyFlag {...props} size={size} />);
        });

        it('should display the flag in `xl`', () => {
          expect(tree.toJSON()).toMatchSnapshot();
        });
      });
    });

    describe('AND a width', () => {
      const width = 28;

      beforeEach(() => {
        tree = renderer.create(<CurrencyFlag {...props} width={width} />);
      });

      it('should display the flag in specified width with calculated height', () => {
        expect(tree.toJSON()).toHaveStyleRule('width', `${width}px`);
        expect(tree.toJSON()).toHaveStyleRule('height', `${width / 1.6}px`);
      });
    });

    describe('AND a height', () => {
      const height = 28;

      beforeEach(() => {
        tree = renderer.create(<CurrencyFlag {...props} height={height} />);
      });

      it('should display the flag in specified height with calculated width', () => {
        expect(tree.toJSON()).toHaveStyleRule('height', `${height}px`);
        expect(tree.toJSON()).toHaveStyleRule('width', `${height * 1.6}px`);
      });
    });

    describe('And no width or height', () => {
      beforeEach(() => {
        tree = renderer.create(<CurrencyFlag currency={currency} />);
      });

      it('should display the flag with default dimensions', () => {
        expect(tree.toJSON()).toMatchSnapshot();
      });
    });
  });

  describe('GIVEN an invalid/unsupported currency', () => {
    const currency = '**INVALID**';

    beforeEach(() => {
      tree = renderer.create(
        <CurrencyFlag currency={currency} size={CurrencyFlagSizes.SMALL} />
      );
    });

    it('should display the empty flag', () => {
      expect(tree.toJSON()).toHaveStyleRule(
        'background-image',
        `url("${flags.$$$}")`
      );
    });
  });
});
