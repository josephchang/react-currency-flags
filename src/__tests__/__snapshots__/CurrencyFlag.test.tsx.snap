// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CurrencyFlag /> GIVEN a valid/supported currency AND a predefined size AND the size \`lg\` should display the flag in \`lg\` 1`] = `
.c0 {
  display: inline-block;
  height: 24px;
  width: 36px;
  background-size: cover;
  background-image: url("data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAADAAAAAgCAMAAABjCgsuAAAAVFBMVEX/AAD/AwP/BQX/CQn/R0f/Skr/S0v/Tk7/cHD/cXH/hYX/lJT/q6v/r6//srL/t7f/xMT/zc3/zs7/z8//0ND/1dX/1tb/2Nj/29v/5ub/9/f////rrEQTAAAAbklEQVQ4y+3Ttw6AMAxF0QsJvRMI7f//kwExICFkjwje5uEMbnDJdhOe8oO3g7bVgSGKBxWwYOVgWasUsmpdhKAITAKJCQohqM+qFoIyPIqwlDbdGADTyKeUA+SKsXrXd86rNj3NytNw4/9AHwc7pig+ByeVQt0AAAAASUVORK5CYII=");
}

<span
  className="c0"
  size="lg"
/>
`;

exports[`<CurrencyFlag /> GIVEN a valid/supported currency AND a predefined size AND the size \`md\` should display the flag in \`md\` 1`] = `
.c0 {
  display: inline-block;
  height: 16px;
  width: 24px;
  background-size: cover;
  background-image: url("data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAADAAAAAgCAMAAABjCgsuAAAAVFBMVEX/AAD/AwP/BQX/CQn/R0f/Skr/S0v/Tk7/cHD/cXH/hYX/lJT/q6v/r6//srL/t7f/xMT/zc3/zs7/z8//0ND/1dX/1tb/2Nj/29v/5ub/9/f////rrEQTAAAAbklEQVQ4y+3Ttw6AMAxF0QsJvRMI7f//kwExICFkjwje5uEMbnDJdhOe8oO3g7bVgSGKBxWwYOVgWasUsmpdhKAITAKJCQohqM+qFoIyPIqwlDbdGADTyKeUA+SKsXrXd86rNj3NytNw4/9AHwc7pig+ByeVQt0AAAAASUVORK5CYII=");
}

<span
  className="c0"
  size="md"
/>
`;

exports[`<CurrencyFlag /> GIVEN a valid/supported currency AND a predefined size AND the size \`sm\` should display the flag in \`sm\` 1`] = `
.c0 {
  display: inline-block;
  height: 10px;
  width: 16px;
  background-size: cover;
  background-image: url("data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAADAAAAAgCAMAAABjCgsuAAAAVFBMVEX/AAD/AwP/BQX/CQn/R0f/Skr/S0v/Tk7/cHD/cXH/hYX/lJT/q6v/r6//srL/t7f/xMT/zc3/zs7/z8//0ND/1dX/1tb/2Nj/29v/5ub/9/f////rrEQTAAAAbklEQVQ4y+3Ttw6AMAxF0QsJvRMI7f//kwExICFkjwje5uEMbnDJdhOe8oO3g7bVgSGKBxWwYOVgWasUsmpdhKAITAKJCQohqM+qFoIyPIqwlDbdGADTyKeUA+SKsXrXd86rNj3NytNw4/9AHwc7pig+ByeVQt0AAAAASUVORK5CYII=");
}

<span
  className="c0"
  size="sm"
/>
`;

exports[`<CurrencyFlag /> GIVEN a valid/supported currency AND a predefined size AND the size \`xl\` should display the flag in \`xl\` 1`] = `
.c0 {
  display: inline-block;
  height: 32px;
  width: 48px;
  background-size: cover;
  background-image: url("data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAADAAAAAgCAMAAABjCgsuAAAAVFBMVEX/AAD/AwP/BQX/CQn/R0f/Skr/S0v/Tk7/cHD/cXH/hYX/lJT/q6v/r6//srL/t7f/xMT/zc3/zs7/z8//0ND/1dX/1tb/2Nj/29v/5ub/9/f////rrEQTAAAAbklEQVQ4y+3Ttw6AMAxF0QsJvRMI7f//kwExICFkjwje5uEMbnDJdhOe8oO3g7bVgSGKBxWwYOVgWasUsmpdhKAITAKJCQohqM+qFoIyPIqwlDbdGADTyKeUA+SKsXrXd86rNj3NytNw4/9AHwc7pig+ByeVQt0AAAAASUVORK5CYII=");
}

<span
  className="c0"
  size="xl"
/>
`;

exports[`<CurrencyFlag /> GIVEN a valid/supported currency And no width or height should display the flag with default dimensions 1`] = `
.c0 {
  display: inline-block;
  height: 16px;
  width: 24px;
  background-size: cover;
  background-image: url("data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAADAAAAAgCAMAAABjCgsuAAAAVFBMVEX/AAD/AwP/BQX/CQn/R0f/Skr/S0v/Tk7/cHD/cXH/hYX/lJT/q6v/r6//srL/t7f/xMT/zc3/zs7/z8//0ND/1dX/1tb/2Nj/29v/5ub/9/f////rrEQTAAAAbklEQVQ4y+3Ttw6AMAxF0QsJvRMI7f//kwExICFkjwje5uEMbnDJdhOe8oO3g7bVgSGKBxWwYOVgWasUsmpdhKAITAKJCQohqM+qFoIyPIqwlDbdGADTyKeUA+SKsXrXd86rNj3NytNw4/9AHwc7pig+ByeVQt0AAAAASUVORK5CYII=");
}

<span
  className="c0"
/>
`;
